### ISSUE TITLE
questions regarding lineage json

### ISSUE THREAD

**Issue by azahnen** [Geonovum/WaU-UC5#1]
This is the lineage I get for `bouwjaar`:

```json
"pathMapping": [
  {
    "path": [
      {
        "segments": [
          "isGerelateerdAan",
          "oorspronkelijkBouwjaar"
        ],
        "references": [
          {
            "kenmerk": "oorspronkelijkBouwjaar",
            "onderwerp": {
              "type": "Pand",
              "identificatie": "0200100000085932"
            }
          }
        ],
        "startNode": {
          "type": "Pand",
          "identificatie": "G0200.42b3d39246840268e0530a0a28492340"
        }
      }
    ]
  }
]
```

1. According to the model the references might contain a `bronregistratie` object, but it is currently (during the last high5) not provided by the orchestration engine. That would be needed to show in the visualization that these are different Pands from BGT and BAG.
2. The startNode may not contain `bronregistratie` object, but it would be needed as well. But I think there should be a reference for `isGerelateerdAan` which could provide it? 
3. There is another case for `heeftAlsAdres` where it seems that the reference for `bevat` is missing. Do I understand it correctly that there should be a reference for each segment?

**Comment by pmaria** [comment]
> This is the lineage I get for `bouwjaar`:
> 
> ```json
> "pathMapping": [
>   {
>     "path": [
>       {
>         "segments": [
>           "isGerelateerdAan",
>           "oorspronkelijkBouwjaar"
>         ],
>         "references": [
>           {
>             "kenmerk": "oorspronkelijkBouwjaar",
>             "onderwerp": {
>               "type": "Pand",
>               "identificatie": "0200100000085932"
>             }
>           }
>         ],
>         "startNode": {
>           "type": "Pand",
>           "identificatie": "G0200.42b3d39246840268e0530a0a28492340"
>         }
>       }
>     ]
>   }
> ]
> ```
> 
> 1. According to the model the references might contain a `bronregistratie` object, but it is currently (during the last high5) not provided by the orchestration engine. That would be needed to show in the visualization that these are different Pands from BGT and BAG.

Yes, we don't really have a solution for this yet. This would require some kind of metadata endpoint per source. As a temporary fix we could misuse the model alias from the mapping, but we would need to think of a more stable solution.

> 2. The startNode may not contain `bronregistratie` object, but it would be needed as well. But I think there should be a reference for `isGerelateerdAan` which could provide it?

Yes adding it to the start node might be a useful extension to the model.

> 3. There is another case for `heeftAlsAdres` where it seems that the reference for `bevat` is missing. Do I understand it correctly that there should be a reference for each segment?

Hmm no, the idea is that only the target data elements of the full path are returned as references. The goal is to only return those references that are actually used in creating a new data element.

**Comment by azahnen** [comment]
@pmaria 
> > 1. According to the model the references might contain a bronregistratie object, but it is currently (during the last high5) not provided by the orchestration engine. That would be needed to show in the visualization that these are different Pands from BGT and BAG.

> Yes, we don't really have a solution for this yet. This would require some kind of metadata endpoint per source. As a temporary fix we could misuse the model alias from the mapping, but we would need to think of a more stable solution.

Yeah, some kind of alias was what I was expecting. Maybe `bronregistratie` is not the right tool here and should be reserved for the actual source metadata. But I think there needs to be some way to differentiate obect types, maybe `onderwerp` could have an additional `source` property with the alias?

> > 3. There is another case for heeftAlsAdres where it seems that the reference for bevat is missing. Do I understand it correctly that there should be a reference for each segment?

> Hmm no, the idea is that only the target data elements of the full path are returned as references. The goal is to only return those references that are actually used in creating a new data element.

Understood, but then I think having only the property names in segments is not very helpful. Without an object type (and source), what is the purpose of the information given?

This is the current visualization for `heeftAlsAdres`, note the `unknown` node:

<img width="1480" alt="Bildschirmfoto 2023-06-13 um 07 45 25" src="https://github.com/Geonovum/WaU-UC5/assets/15887231/da7d9f87-d22c-469f-b07c-d23c398d15fa">

<br/>This is the lineage for `heeftAlsAdres `:

```json
{
  "path": [
    {
      "segments": [
        "isGerelateerdAan",
        "bevat",
        "heeftAlsHoofdadres"
      ],
      "references": [
        {
          "kenmerk": "heeftAlsHoofdadres",
          "onderwerp": {
            "type": "Verblijfsobject",
            "identificatie": "0200010000130332"
          }
        },
        {
          "kenmerk": "heeftAlsHoofdadres",
          "onderwerp": {
            "type": "Verblijfsobject",
            "identificatie": "0200010000130331"
          }
        }
      ],
      "startNode": {
        "type": "Pand",
        "identificatie": "G0200.42b3d39246840268e0530a0a28492340"
      }
    }
  ]
}
```
